espressif/vscode-esp-idf-extension
# Bugfix/debugging docs

## Description

Update documentation regarding debugging using other extensions and now using Eclipse CDT GDB

Fixes #918
Fixes #1164 
Fixes #1138 
Fixes #1168 
Fixes #1177

## Type of change
- This change requires a documentation update
- New feature

## Steps to test this pull request

1. Create a new ESP-IDF Project
2. Build flash the project
3. Use the CDT debug configuration to start debugging.

```json
 {
   "configurations": [
     {
       "type": "gdbtarget",
       "request": "attach",
       "name": "Eclipse CDT Remote",
       "program": "${workspaceFolder}/build/${command:espIdf.getProjectName}.elf",
       "initCommands": [
         "set remote hardware-watchpoint-limit 2",
         "mon reset halt",
         "maintenance flush register-cache",
         "thb app_main",
       ],
       "gdb": "${command:espIdf.getXtensaGdb}",
       "target": {
         "port": "3333"
       },
     }
   ]
 }
```
4. Perform debug tasks. Read the updated debugging tutorial for example.

- Expected behaviour:

New debugging should work faster and better than previous debug adapter.

- Expected output:

Debugging experience should be the same as debugging tutorial.

## How has this been tested?

Test running the debugging tutorial.

**Test Configuration**:
* ESP-IDF Version: 5.0
* OS (Windows,Linux and macOS): macOS

## Dependent components impacted by this PR:

- ESP-Debug Adapter

## Checklist
